src/message-handler/src/HiveMindHostApiRequestHandler.cpp

```diff
@@ -20,11 +21,48 @@ bool HiveMindHostApiRequestHandler::handleRequest(const MessageDTO& message) {
 }
 
 bool HiveMindHostApiRequestHandler::handleHiveMindHostApiRequest(
-    const MessageDTO& message, const HiveMindHostApiRequestDTO& request) {
+    uint16_t requestId, const MessageDTO& message, const HiveMindHostApiRequestDTO& request) {
+
     if (std::holds_alternative<BytesDTO>(request.getRequest())) {
         return m_hostQueue.push(message);
     }
 
+    if (const auto* neighborReq = std::get_if<GetNeighborRequestDTO>(&request.getRequest())) {
+        uint16_t id = neighborReq->getNeighborId();
+        std::optional<RelativePosition> posOpt = m_interloc.getRobotPosition(id);
+
+        std::optional<NeighborPositionDTO> posOptDTO = {};
+        if (posOpt) {
+
+            const RelativePosition& pos = posOpt.value();
+            posOptDTO = NeighborPositionDTO(pos.m_distance, pos.m_relativeOrientation,
+                                            pos.m_isInLineOfSight);
+        }
+
+        GetNeighborResponseDTO neighborResp(id, posOptDTO);
+        HiveMindHostApiResponseDTO hmResp(neighborResp);
+        ResponseDTO resp(requestId, hmResp);
+        MessageDTO msg(m_bsp.getUUId(), message.getSourceId(), resp);
+        return m_hostQueue.push(msg);
+    }
+
+    if (std::holds_alternative<GetNeighborsListRequestDTO>(request.getRequest())) {
+
+        GetNeighborsListResponseDTO neighborResp(NULL, 0);
+        const PositionsTable& posTable = m_interloc.getPositionsTable();
+
+        neighborResp.setRawNeighborsLength(posTable.m_positionsLength);
+        auto& rawNeighbors = neighborResp.getRawNeighbors();
+        for (uint16_t i = 0; i < neighborResp.getNeighborsLength(); i++) {
+            rawNeighbors[i] = posTable.m_positions[i].m_robotId;
```

Comment:
If a robot is not in LOS but is still present in the table because it is still in range of the DW, is it considered a neighbour?

Reply:
As of now yes it does. If we found the need to only have the ones in LOS, we could add a field in the request  to filter and send only those in LOS.